plugins/linux-v4l2/v4l2-decoder.c

```diff
@@ -78,7 +78,7 @@ void v4l2_destroy_decoder(struct v4l2_decoder *decoder)
 	}
 
 	if (decoder->context) {
-		avcodec_close(decoder->context);
+		// avcodec_close(decoder->context);
 		avcodec_free_context(&decoder->context);
```

Comment:
In what version of FFmpeg does this break? How far back is this syntax permitted/recommended? We have to support as far back as FFmpeg 4.4.2 due to still supporting Ubuntu 22.04.

Reply:
`
#define FF_API_AVCODEC_CLOSE       (LIBAVCODEC_VERSION_MAJOR < 62)
#if FF_API_AVCODEC_CLOSE
/**
 * Close a given AVCodecContext and free all the data associated with it
 * (but not the AVCodecContext itself).
 *
 * Calling this function on an AVCodecContext that hasn't been opened will free
 * the codec-specific data allocated in avcodec_alloc_context3() with a non-NULL
 * codec. Subsequent calls will do nothing.
 *
 * @deprecated Do not use this function. Use avcodec_free_context() to destroy a
 * codec context (either open or closed). Opening and closing a codec context
 * multiple times is not supported anymore -- use multiple codec contexts
 * instead.
 */
attribute_deprecated
int avcodec_close(AVCodecContext *avctx);
#endif`

Reply:
https://ffmpeg.org/doxygen/trunk/group__lavc__core.html#gab7f632082a10fcb051fa3e2db7f86942
Reference is given above.